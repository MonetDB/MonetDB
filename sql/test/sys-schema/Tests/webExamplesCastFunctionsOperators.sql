select cast(123 as varchar(10));
select convert(123, decimal(10,3));

select cast(true as smallint);
select cast(42 as int);
select cast(123.45 as real);
select cast('123.45' as double precision);
select cast(23.45 as decimal(5,2));    -- precision of 5 digits of which 2 decimal digits

select cast('2020-07-29' as date);
select cast('17:44:59' as time);
select cast('17:44:59.123456' as time);
select cast('2020-07-29 17:44:59' as timestamp);
select cast('2020-07-29T17:44:59' as timestamp);
select cast('2020-07-29 17:44:59.123456' as timestamp);
select cast('17:44:59.321+01:30' as timetz);
select cast('2020-07-29 17:44:59.321+01:30' as timestamptz);	-- produces wrong TZ +1:00 instead of +1:30
select cast('1234' as interval month);
select cast('86400.123' as interval second);

select cast('abcd' as blob);
select cast('abcde' as clob);
select cast('192.168.1.5/24' as inet);
select cast(r'{"a":[1,2,4]}' as json);
select cast('https://www.monetdb.org/Home' as url);
select cast('e31960fb-dc8b-452d-ab30-b342723e756a' as uuid);

-- or using convert instead of cast:
select convert('a4cd' , blob);
select convert('abcde' , clob);
select convert('192.168.1.5/24' , inet);
select convert(r'{"a":[1,2,4]}' , json);
select convert('https://www.monetdb.org/Home' , url);
select convert('e31960fb-dc8b-452d-ab30-b342723e756a' , uuid);

-- using prefix operators
select x'abcd';
select blob 'abcd';
select clob 'abcde';	-- this fails on Oct2020
select cast('abcde' as clob);
select inet '192.168.1.5/24';
select json '{"a":[1,2,4]}';
select url 'https://www.monetdb.org/Home';
select uuid 'e31960fb-dc8b-452d-ab30-b342723e756a';
select E'EA\fB\tC\n\\Z';
select e'eA\fB\tC\n\\Z';
select R'RA\fB\tC\n\\Z';
select r'rA\fB\tC\n\\Z';

