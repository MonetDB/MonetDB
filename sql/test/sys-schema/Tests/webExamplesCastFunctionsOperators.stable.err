stderr of test 'webExamplesCastFunctionsOperators` in directory 'sql/test/sys-schema` itself:


# 19:23:26 >  
# 19:23:26 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-10408" "--port=37113"
# 19:23:26 >  


# 19:01:20 >  
# 19:01:20 >  "Done."
# 19:01:20 >  

