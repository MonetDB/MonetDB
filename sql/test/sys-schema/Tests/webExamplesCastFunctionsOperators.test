query T rowsort
select cast(123 as varchar(10))
----
123

query R rowsort
select convert(123, decimal(10,3))
----
123.000

query I rowsort
select cast(true as smallint)
----
1

query I rowsort
select cast(42 as int)
----
42

query R rowsort
select cast(123.45 as real)
----
123.450

query R rowsort
select cast('123.45' as double precision)
----
123.450

query R rowsort
select cast(23.45 as decimal(5,2))
----
23.450

query T rowsort
select cast('2020-07-29' as date)
----
2020-07-29

query T rowsort
select cast('17:44:59' as time)
----
17:44:59

query T rowsort
select cast('17:44:59.123456' as time)
----
17:44:59

query T rowsort
select cast('2020-07-29 17:44:59' as timestamp)
----
2020-07-29 17:44:59

query T rowsort
select cast('2020-07-29T17:44:59' as timestamp)
----
2020-07-29 17:44:59

query T rowsort
select cast('2020-07-29 17:44:59.123456' as timestamp)
----
2020-07-29 17:44:59.123456

query T rowsort
select cast('17:44:59.321+01:30' as timetz)
----
16:14:59.321000

query T rowsort
select cast('2020-07-29 17:44:59.321+01:30' as timestamptz)
----
2020-07-29 16:14:59.321000

query T rowsort
select cast('1234' as interval month)
----
1234

query T rowsort
select cast('86400.123' as interval second)
----
1 day, 0:00:00

query T rowsort
select cast('abcd' as blob)
----
ABCD

query T rowsort
select cast('abcde' as clob)
----
abcde

query T rowsort
select cast('192.168.1.5/24' as inet)
----
192.168.1.5/24

query T rowsort
select cast(r'{"a":[1,2,4]}' as json)
----
{"a":[1,2,4]}

query T rowsort
select cast('https://www.monetdb.org/Home' as url)
----
https://www.monetdb.org/Home

query T rowsort
select cast('e31960fb-dc8b-452d-ab30-b342723e756a' as uuid)
----
e31960fb-dc8b-452d-ab30-b342723e756a

query T rowsort
select convert('a4cd' , blob)
----
A4CD

query T rowsort
select convert('abcde' , clob)
----
abcde

query T rowsort
select convert('192.168.1.5/24' , inet)
----
192.168.1.5/24

query T rowsort
select convert(r'{"a":[1,2,4]}' , json)
----
{"a":[1,2,4]}

query T rowsort
select convert('https://www.monetdb.org/Home' , url)
----
https://www.monetdb.org/Home

query T rowsort
select convert('e31960fb-dc8b-452d-ab30-b342723e756a' , uuid)
----
e31960fb-dc8b-452d-ab30-b342723e756a

query T rowsort
select x'abcd'
----
ABCD

query T rowsort
select blob 'abcd'
----
ABCD

statement error
select clob 'abcde'

query T rowsort
select cast('abcde' as clob)
----
abcde

query T rowsort
select inet '192.168.1.5/24'
----
192.168.1.5/24

query T rowsort
select json '{"a":[1,2,4]}'
----
{"a":[1,2,4]}

query T rowsort
select url 'https://www.monetdb.org/Home'
----
https://www.monetdb.org/Home

query T rowsort
select uuid 'e31960fb-dc8b-452d-ab30-b342723e756a'
----
e31960fb-dc8b-452d-ab30-b342723e756a

query T rowsort
select E'EA\fB\tC\n\\Z'
----
EA@B@C@\Z

query T rowsort
select e'eA\fB\tC\n\\Z'
----
eA@B@C@\Z

query T rowsort
select R'RA\fB\tC\n\\Z'
----
RA\fB\tC\n\\Z

query T rowsort
select r'rA\fB\tC\n\\Z'
----
rA\fB\tC\n\\Z


