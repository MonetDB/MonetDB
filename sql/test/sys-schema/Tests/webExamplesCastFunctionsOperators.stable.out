stdout of test 'webExamplesCastFunctionsOperators` in directory 'sql/test/sys-schema` itself:


# 19:01:20 >  
# 19:01:20 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6858" "--port=39255"
# 19:01:20 >  

#select cast(123 as varchar(10));
% .%1 # table_name
% %1 # name
% varchar # type
% 3 # length
[ "123"	]
#select convert(123, decimal(10,3));
% .%1 # table_name
% %1 # name
% decimal # type
% 12 # length
[ 123.000	]
#select cast(true as smallint);
% .%1 # table_name
% %1 # name
% smallint # type
% 1 # length
[ 1	]
#select cast(42 as int);
% .%2 # table_name
% %2 # name
% int # type
% 2 # length
[ 42	]
#select cast(123.45 as real);
% .%2 # table_name
% %2 # name
% real # type
% 15 # length
[ 123.45	]
#select cast('123.45' as double precision);
% .%1 # table_name
% %1 # name
% double # type
% 24 # length
[ 123.45	]
#select cast(23.45 as decimal(5,2));    -- precision of 5 digits of which 2 decimal digits
% .%1 # table_name
% %1 # name
% decimal # type
% 7 # length
[ 23.45	]
#select cast('2020-07-29' as date);
% .%2 # table_name
% %2 # name
% date # type
% 10 # length
[ 2020-07-29	]
#select cast('17:44:59' as time);
% .%2 # table_name
% %2 # name
% time # type
% 8 # length
[ 17:44:59	]
#select cast('17:44:59.123456' as time);
% .%2 # table_name
% %2 # name
% time # type
% 8 # length
[ 17:44:59	]
#select cast('2020-07-29 17:44:59' as timestamp);
% .%2 # table_name
% %2 # name
% timestamp # type
% 26 # length
[ 2020-07-29 17:44:59.000000	]
#select cast('2020-07-29T17:44:59' as timestamp);
% .%2 # table_name
% %2 # name
% timestamp # type
% 26 # length
[ 2020-07-29 17:44:59.000000	]
#select cast('2020-07-29 17:44:59.123456' as timestamp);
% .%2 # table_name
% %2 # name
% timestamp # type
% 26 # length
[ 2020-07-29 17:44:59.123456	]
#select cast('17:44:59.321+01:30' as timetz);
% .%1 # table_name
% %1 # name
% timetz # type
% 21 # length
[ 17:14:59.321000+01:00	]
#select cast('2020-07-29 17:44:59.321+01:30' as timestamptz);	-- produces wrong TZ +1:00 instead of +1:30
% .%1 # table_name
% %1 # name
% timestamptz # type
% 32 # length
[ 2020-07-29 17:14:59.321000+01:30	]
#select cast('1234' as interval month);
% .%1 # table_name
% %1 # name
% month_interval # type
% 4 # length
[ 1234	]
#select cast('86400.123' as interval second);
% .%1 # table_name
% %1 # name
% sec_interval # type
% 9 # length
[ 86400.123	]
#select cast('abcd' as blob);
% .%1 # table_name
% %1 # name
% blob # type
% 0 # length
[ ABCD	]
#select cast('abcde' as clob);
% .%2 # table_name
% %2 # name
% clob # type
% 5 # length
[ "abcde"	]
#select cast('192.168.1.5/24' as inet);
% .%1 # table_name
% %1 # name
% inet # type
% 0 # length
[ 192.168.1.5/24	]
#select cast(r'{"a":[1,2,4]}' as json);
% .%1 # table_name
% %1 # name
% json # type
% 13 # length
[ "{\"a\":[1,2,4]}"	]
#select cast('https://www.monetdb.org/Home' as url);
% .%1 # table_name
% %1 # name
% url # type
% 28 # length
[ https://www.monetdb.org/Home	]
#select cast('e31960fb-dc8b-452d-ab30-b342723e756a' as uuid);
% .%1 # table_name
% %1 # name
% uuid # type
% 36 # length
[ e31960fb-dc8b-452d-ab30-b342723e756a	]
#select convert('a4cd' , blob);
% .%1 # table_name
% %1 # name
% blob # type
% 0 # length
[ A4CD	]
#select convert('abcde' , clob);
% .%2 # table_name
% %2 # name
% clob # type
% 5 # length
[ "abcde"	]
#select convert('192.168.1.5/24' , inet);
% .%1 # table_name
% %1 # name
% inet # type
% 0 # length
[ 192.168.1.5/24	]
#select convert(r'{"a":[1,2,4]}' , json);
% .%1 # table_name
% %1 # name
% json # type
% 13 # length
[ "{\"a\":[1,2,4]}"	]
#select convert('https://www.monetdb.org/Home' , url);
% .%1 # table_name
% %1 # name
% url # type
% 28 # length
[ https://www.monetdb.org/Home	]
#select convert('e31960fb-dc8b-452d-ab30-b342723e756a' , uuid);
% .%1 # table_name
% %1 # name
% uuid # type
% 36 # length
[ e31960fb-dc8b-452d-ab30-b342723e756a	]
#select x'abcd';
% .%2 # table_name
% %2 # name
% blob # type
% 0 # length
[ ABCD	]
#select blob 'abcd';
% .%2 # table_name
% %2 # name
% blob # type
% 0 # length
[ ABCD	]
#select clob 'abcde';
% .%2 # table_name
% %2 # name
% clob # type
% 5 # length
[ "abcde"	]
#select cast('abcde' as clob);
% .%2 # table_name
% %2 # name
% clob # type
% 5 # length
[ "abcde"	]
#select inet '192.168.1.5/24';
% .%2 # table_name
% %2 # name
% inet # type
% 0 # length
[ 192.168.1.5/24	]
#select json '{"a":[1,2,4]}';
% .%2 # table_name
% %2 # name
% json # type
% 13 # length
[ "{\"a\":[1,2,4]}"	]
#select url 'https://www.monetdb.org/Home';
% .%2 # table_name
% %2 # name
% url # type
% 28 # length
[ https://www.monetdb.org/Home	]
#select uuid 'e31960fb-dc8b-452d-ab30-b342723e756a';
% .%2 # table_name
% %2 # name
% uuid # type
% 36 # length
[ e31960fb-dc8b-452d-ab30-b342723e756a	]
#select E'EA\fB\tC\n\\Z';
% .%2 # table_name
% %2 # name
% char # type
% 9 # length
[ "EA\014B\tC\n\\Z"	]
#select e'eA\fB\tC\n\\Z';
% .%2 # table_name
% %2 # name
% char # type
% 9 # length
[ "eA\014B\tC\n\\Z"	]
#select R'RA\fB\tC\n\\Z';
% .%2 # table_name
% %2 # name
% char # type
% 13 # length
[ "RA\\fB\\tC\\n\\\\Z"	]
#select r'rA\fB\tC\n\\Z';
% .%2 # table_name
% %2 # name
% char # type
% 13 # length
[ "rA\\fB\\tC\\n\\\\Z"	]

# 19:01:20 >  
# 19:01:20 >  "Done."
# 19:01:20 >  

